### ISSUE TITLE
Respect signature of metaclass's `__call__`

### ISSUE THREAD

**Issue by taminomara** [astral-sh/ty#2288]
Another feature request similar to #281: respect signature of `__call__` defined on a metaclass.

```python
class Meta(type):
    def __call__(cls, arg: int) -> str: ...

class T(metaclass=Meta):
    pass

# Inferred type: `T`, should be `str`.
t = T()  # Should report missing argument `arg`.
```

**Comment by carljm** [comment]
Confirmed fixed on main.
